File: C++/Exceptions.cpp
#include "Exceptions.h"
using namespace SimpleLion;

FilesystemException::FilesystemException(std::string what) {
	_what = what;
	_dirpath = "";
}

FilesystemException::FilesystemException(std::string what, std::string dirpath) {
	_what = what;
	_dirpath = dirpath;
}

FilesystemException::~FilesystemException() throw() {
	_what = "";
	_dirpath = "";
}

const char* FilesystemException::what() const throw() {
	std::string msg = "";
	if(_dirpath != "") {
		msg += _dirpath;
		msg += ": ";
	}
	msg += _what;
	return msg.c_str();
}

FileException::FileException(std::string what) {
	_what = what;
	_filepath = "";
}

FileException::FileException(std::string what, std::string filepath) {
	_what = what;
	_filepath = filepath;
}

FileException::~FileException() throw() {
	_what = "";
	_filepath = "";
}

const char* FileException::what() const throw() {
	std::string msg = "";
	if(_filepath != "") {
		msg += _filepath;
		msg += ": ";
	}
	msg += _what;
	return msg.c_str();
}

File: Ruby/lib/exceptions.rb
module SimpleLion
	
	##
	# This is an exception that is raised when there is a problem with the localization file - mostly when it does not exist
	
	class FileException < Exception
		
		##
		# This constructor initializes the exception with error message that states which +file+ the script was trying to open and what happened - by +msg+
		
		def initialize file, msg
			super(file.to_s + ": " + msg.to_s)
		end
	end
	
	##
	# This is an exception that is raised when there is a problem with the directory - it is either unaccessible, cannot be created or does not have any SLLF localization file inside
	
	class FilesystemException < Exception
		
		##
		# This constructor initializes the exception with error message that states which +dir+ the script was trying to process and what happened - by +msg+
		
		def initialize dir, msg
			super(dir.to_s + ": " + msg.to_s)
		end
	end
end
